pilot/pkg/security/model/authentication.go

```diff
@@ -217,10 +241,17 @@ func ApplyToCommonTLSContext(tlsContext *tls.CommonTlsContext, metadata *model.N
 			CaCertificatePath: metadata.TLSServerRootCert,
 		}
 
+		var matchSAN []*matcher.StringMatcher
```

Comment:
The `subjectAltNames` is always needed, you can just:
```go
matchSAN := util.StringToExactMatch(subjectAltNames)
if len(trustDomainAliases) > 0 {
  matchSAN = append(matchSAN, util.StringToPrefixMatch(trustDomainAliases))
}
```

Reply:
One issue with this is that we also need to appaend spiffe.URIPrefix to all entries, where do you suggest we do that